# QUESTION 4 - STILL WRONG

Which was the pick up day with the longest trip distance? Only consider trips with trip_distance less than 100 miles. (1 point) <br>
a. 2025-11-14 <br>
b. 2025-11-20 <br>
c. 2025-11-23 <br>
d. 2025-11-25


In [1]:
#download dataset
!wget https://d37ci6vzurychx.cloudfront.net/trip-data/green_tripdata_2025-11.parquet

--2026-01-24 15:43:38--  https://d37ci6vzurychx.cloudfront.net/trip-data/green_tripdata_2025-11.parquet
Resolving d37ci6vzurychx.cloudfront.net (d37ci6vzurychx.cloudfront.net)... 99.84.231.41, 99.84.231.48, 99.84.231.38, ...
Connecting to d37ci6vzurychx.cloudfront.net (d37ci6vzurychx.cloudfront.net)|99.84.231.41|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1164775 (1.1M) [binary/octet-stream]
Saving to: ‘green_tripdata_2025-11.parquet’

green_tripdata_2025 100%[===================>]   1.11M  7.35MB/s    in 0.2s    

2026-01-24 15:43:38 (7.35 MB/s) - ‘green_tripdata_2025-11.parquet’ saved [1164775/1164775]



In [2]:
import pandas as pd
df = pd.read_parquet("green_tripdata_2025-11.parquet")

In [3]:
#checking the data sample
df.head(2)

VendorID lpep_pickup_datetime lpep_dropoff_datetime store_and_fwd_flag  \
0         2  2025-11-01 00:34:48   2025-11-01 00:41:39                  N   
1         2  2025-11-01 00:18:52   2025-11-01 00:24:27                  N   

   RatecodeID  PULocationID  DOLocationID  passenger_count  trip_distance  \
0         1.0            74            42              1.0           0.74   
1         1.0            74            42              2.0           0.95   

   fare_amount  ...  mta_tax  tip_amount  tolls_amount  ehail_fee  \
0          7.2  ...      0.5        1.94           0.0        NaN   
1          7.2  ...      0.5        0.00           0.0        NaN   

   improvement_surcharge  total_amount  payment_type  trip_type  \
0                    1.0         11.64           1.0        1.0   
1                    1.0          9.70           2.0        1.0   

   congestion_surcharge  cbd_congestion_fee  
0                   0.0                 0.0  
1                   0.0                 0.0  

[2 rows x 21 columns]

In [9]:
# filter for the distance below 100
df_2 = df[df["trip_distance"] <= 100]

In [10]:
df_2

VendorID lpep_pickup_datetime lpep_dropoff_datetime store_and_fwd_flag  \
0             2  2025-11-01 00:34:48   2025-11-01 00:41:39                  N   
1             2  2025-11-01 00:18:52   2025-11-01 00:24:27                  N   
2             2  2025-11-01 01:03:14   2025-11-01 01:15:24                  N   
3             2  2025-11-01 00:10:57   2025-11-01 00:24:53                  N   
4             1  2025-11-01 00:03:48   2025-11-01 00:19:38                  N   
...         ...                  ...                   ...                ...   
46907         2  2025-11-30 19:58:34   2025-11-30 20:14:28               None   
46908         2  2025-11-30 19:34:00   2025-11-30 19:46:00               None   
46909         2  2025-11-30 21:46:46   2025-11-30 22:17:55               None   
46910         2  2025-11-30 21:00:00   2025-11-30 21:15:00               None   
46911         2  2025-11-30 23:26:00   2025-11-30 23:42:00               None   

       RatecodeID  PULocationID  DOLocationID  passenger_count  trip_distance  \
0             1.0            74            42              1.0           0.74   
1             1.0            74            42              2.0           0.95   
2             1.0            83           160              1.0           2.19   
3             1.0           166           127              1.0           5.44   
4             1.0           166           262              1.0           3.20   
...           ...           ...           ...              ...            ...   
46907         NaN            59            51              NaN           8.50   
46908         NaN            74           151              NaN           1.73   
46909         NaN            33           163              NaN           7.52   
46910         NaN            16            95              NaN           5.61   
46911         NaN            65            87              NaN           2.91   

       fare_amount  ...  tip_amount  tolls_amount  ehail_fee  \
0             7.20  ...        1.94           0.0        NaN   
1             7.20  ...        0.00           0.0        NaN   
2            13.50  ...        5.00           0.0        NaN   
3            24.70  ...        0.50           0.0        NaN   
4            18.40  ...        1.00           0.0        NaN   
...            ...  ...         ...           ...        ...   
46907        33.22  ...        0.00           0.0        NaN   
46908        13.86  ...        0.77           0.0        NaN   
46909        38.42  ...        1.00           0.0        NaN   
46910        24.67  ...        0.00           0.0        NaN   
46911        14.46  ...        3.89           0.0        NaN   

       improvement_surcharge  total_amount  payment_type  trip_type  \
0                        1.0         11.64           1.0        1.0   
1                        1.0          9.70           2.0        1.0   
2                        1.0         21.00           1.0        1.0   
3                        1.0         27.70           1.0        1.0   
4                        1.0         24.65           1.0        1.0   
...                      ...           ...           ...        ...   
46907                    1.0         34.72           NaN        NaN   
46908                    1.0         16.13           NaN        NaN   
46909                    1.0         44.42           NaN        NaN   
46910                    1.0         26.17           NaN        NaN   
46911                    1.0         23.35           NaN        NaN   

       congestion_surcharge  cbd_congestion_fee  filter_distance  
0                      0.00                0.00             True  
1                      0.00                0.00             True  
2                      0.00                0.00             True  
3                      0.00                0.00             True  
4                      2.75                0.00             True  
...                     ...          

In [17]:
#Sort the trip distance with descending order
df_2[["lpep_pickup_datetime","trip_distance"]].sort_values(by='trip_distance', ascending=False).head(3)

lpep_pickup_datetime  trip_distance
18867  2025-11-14 15:36:27          88.03
27301  2025-11-20 12:28:02          73.84
31551  2025-11-23 10:12:18          45.26

Longest trip below 100 miles is 88.03 miles, which has pickup date at 2025-11-14, <br>
so the answer is <b> 2025-11-14 (a) </b>